public/components/common/search/direct_search.tsx

```diff
@@ -220,31 +243,32 @@ export const DirectSearch = (props: any) => {
   useEffect(() => {
     // cancel direct query
     if (!pollingResult) return;
-    const { status, datarows } = pollingResult;
+    const { status: anyCaseStatus, datarows, error } = pollingResult;
+    const status = anyCaseStatus?.toLowerCase();
 
     if (status === DirectQueryLoadingStatus.SUCCESS || datarows) {
-      // stop polling
-      stopPolling();
-      setIsQueryRunning(false);
+      stopPollingWithStatus(status);
+      // update page with data
+      dispatchOnGettingHis(pollingResult, '');
+    } else if (status === DirectQueryLoadingStatus.FAILED) {
```

Comment:
Should we also handle the status canceled here?

Reply:
Canceled is handled separately (in direct_query_running.tsx), as that's where cancel is triggered, and so the polling is stopped from there directly.

Reply:
Ideally we should keep it in a single place if possible. Not a hard requirement in the PR 😄